src/ttyname.c

```diff
@@ -131,7 +131,7 @@ get_process_ttyname(char *name, size_t namelen)
     }
     errno = ENOENT;
     if (rc != -1) {
-	if ((dev_t)ki_proc->sudo_kp_tdev != (dev_t)-1) {
```

Comment:
This cast is needed on systems like OpenBSD and NetBSD where p_tdev is uin32_t, not dev_t.